solution/0500-0599/0503.Next Greater Element II/Solution.ts

```diff
@@ -0,0 +1,12 @@
+function nextGreaterElements(nums: number[]): number[] {
+    let stack: number[] = [], len = nums.length;
+    let res: number[] = new Array(len).fill(-1);
+    for (let i = 0; i < 2 * len - 1; i++) {
+        while (stack.length !== 0 && nums[stack[stack.length - 1]] < nums[i % len]) {
+            res[stack[stack.length - 1]] = nums[i % len];
+            stack.pop();
+        }
+        stack.push(i % len);
```

Comment:
`i % len` 只需要计算一次即可。

```suggestion
	const j = i % len;
        while (stack.length !== 0 && nums[stack[stack.length - 1]] < nums[j]) {
            res[stack[stack.length - 1]] = nums[j];
            stack.pop();
        }
        stack.push(j);
```